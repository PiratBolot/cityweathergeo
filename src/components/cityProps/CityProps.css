.city_name {
    font-size: 24px;
    font-weight: bold;
    padding: 0 5px;
    color: #435257;
}

.weather_frame_now {
    flex-grow: 1;
    padding: 0px 7px;
}

.date {
    font-weight: bold;
    padding-top: 10px;
}

.weather_now {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    padding: 10px 15px 10px 10px;
}

.weather_description {
    font-size: large;
    text-align: start;
    color: #435257;
    font-weight: bold;
    padding: 0px 0px;
}