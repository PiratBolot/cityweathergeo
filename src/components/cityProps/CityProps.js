import React from 'react';
import './CityProps.css'
import RemoveCityButton from "./removeCityButton/RemoveCityButton";
import Temperature from "./temperature/Temperature";
import WeatherIcon from "./weatherIcon/WeatherIcon";

class CityProps extends React.Component {
    render() {
        return (
            <div>
                {
                    this.props.isCurrentCity ?
                        <div className="weather_frame_now">
                            <div className="city_name">{this.props.city}</div>
                            <div className="date">{new Date().toDateString()}</div>
                            <div className="weather_now">
                                <Temperature temp={this.props.data.main.temp}/>
                                <WeatherIcon icon={this.props.data.weather[0].icon}/>
                                <div className="weather_description">{this.props.data.weather[0].description}</div>
                            </div>
                        </div>
                        :
                        <div className="weather_now">
                            <div className="city_name">
                                {this.props.city}
                            </div>
                            {
                                this.props.data ?
                                    <Temperature temp={this.props.data.main.temp}/> : ""
                            }
                            {
                                this.props.data ?
                                    <WeatherIcon icon={this.props.data.weather[0].icon}/> : ""
                            }
                            <RemoveCityButton city={this.props.city}/>
                        </div>
                }
            </div>
        )
    }
}

export default CityProps;