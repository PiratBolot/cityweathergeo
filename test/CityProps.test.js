import React from 'react';
import { shallow } from '../enzyme/enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CityProps from "../src/components/cityProps/CityProps";
import apiResponseExample from "./inputDataExample/apiResponseExample";

describe('City Properties', () => {
    const apiResponse = apiResponseExample();

    it('should render Current City properties correctly', () => {
        const cityName = "Syktyvkar";
        const output = shallow(
            <CityProps isCurrentCity={true} city={cityName} data={apiResponse}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('should render Tracked City properties correctly', () => {
        const cityName = "Moscow";
        const output = shallow(
            <CityProps isCurrentCity={false} city={cityName} data={apiResponse} />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});