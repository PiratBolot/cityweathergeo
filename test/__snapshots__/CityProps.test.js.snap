// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City Properties should render Current City properties correctly 1`] = `
<div>
  <div
    className="weather_frame_now"
  >
    <div
      className="city_name"
    >
      Syktyvkar
    </div>
    <div
      className="date"
    >
      Wed Dec 11 2019
    </div>
    <div
      className="weather_now"
    >
      <Temperature
        temp={276.15}
      />
      <WeatherIcon
        icon="04n"
      />
      <div
        className="weather_description"
      >
        пасмурно
      </div>
    </div>
  </div>
</div>
`;

exports[`City Properties should render Tracked City properties correctly 1`] = `
<div>
  <div
    className="weather_now"
  >
    <div
      className="city_name"
    >
      Moscow
    </div>
    <Temperature
      temp={276.15}
    />
    <WeatherIcon
      icon="04n"
    />
    <Connect(RemoveCityButton)
      city="Moscow"
    />
  </div>
</div>
`;
